lib/engine/game/g_18_ireland/step/merge.rb

```diff
@@ -248,13 +249,39 @@ def mergeable_type(entity)
             end
           end
 
+          def corporation_connected?(c1, c2)
+            # Are corporations connected (blocking by tokens doesn't matter)
+            unless @corp_connectivity[c1]
+              new_home = c1.tokens.first.city
+              connected = []
+              visited = {}
+              new_home.walk(skip_track: :narrow, tile_type: @game.class::TILE_TYPE) do |path, _, _|
+                next if visited[path]
+
+                visited[path] = true
+
+                path.nodes.each do |p_node|
+                  next unless p_node.city?
+                  next if visited[p_node]
+
+                  visited[p_node] = true
+
+                  a = p_node.tokens.select { |t| t&.corporation&.type == :minor }.map(&:corporation)
+                  connected.concat(a)
```

Comment:
this looks like it's gonna be expensive

p_node.tokens.each do |...|
connected << corp